Error while checking build/O2/fullCI for da37872a1d11721fadbab47ae60831b4abf8e692 at 2023-02-27 16:26:
```
## sw/BUILD/o2checkcode-latest/log
/sw/SOURCES/O2/10812-slc8_x86-64/0/GPU/Common/test/testGPUsortCUDA.cu:22:10: error: 'boost/test/unit_test.hpp' file not found [clang-diagnostic-error]
/sw/SOURCES/O2/10812-slc8_x86-64/0/GPU/GPUTracking/TRDTracking/GPUTRDTracker.cxx:37:10: error: 'omp.h' file not found [clang-diagnostic-error]
/sw/SOURCES/O2/10812-slc8_x86-64/0/GPU/GPUTracking/Base/GPUReconstruction.cxx:37:10: error: 'omp.h' file not found [clang-diagnostic-error]
/sw/SOURCES/O2/10812-slc8_x86-64/0/GPU/GPUTracking/Base/GPUReconstructionCPU.cxx:45:10: error: 'omp.h' file not found [clang-diagnostic-error]
ors/ITSMFT/ITS/include" "-I" "/sw/SOURCES/O2/10812-slc8_x86-64/0/Steer/include" "-I" "/sw/SOURCES/O2/10812-slc8_x86-64/0/Generators/include" "-I" "/sw/SOURCES/O2/10812-slc8_x86-64/0/Detectors/ITSMFT/ITS/tracking/GPU" "-D" "NDEBUG" "-internal-isystem" "/usr/lib/gcc/x86_64-redhat-linux/8/../../../../include/c++/8" "-internal-isystem" "/usr/lib/gcc/x86_64-redhat-linux/8/../../../../include/c++/8/x86_64-redhat-linux" "-internal-isystem" "/usr/lib/gcc/x86_64-redhat-linux/8/../../../../include/c++/8/backward" "-internal-isystem" "/usr/lib/gcc/x86_64-redhat-linux/8/../../../../include/c++/8" "-internal-isystem" "/usr/lib/gcc/x86_64-redhat-linux/8/../../../../include/c++/8/x86_64-redhat-linux" "-internal-isystem" "/usr/lib/gcc/x86_64-redhat-linux/8/../../../../include/c++/8/backward" "-internal-isystem" "/sw/slc8_x86-64/o2codechecker/v13.0-local2/lib/clang/13.0.0/include" "-internal-isystem" "/usr/local/include" "-internal-isystem" "/usr/lib/gcc/x86_64-redhat-linux/8/../../../../x86_64-redhat-linux/include" "-internal-externc-isystem" "/include" "-internal-externc-isystem" "/usr/include" "-internal-isystem" "/sw/slc8_x86-64/o2codechecker/v13.0-local2/lib/clang/13.0.0/include" "-internal-isystem" "/usr/local/include" "-internal-isystem" "/usr/lib/gcc/x86_64-redhat-linux/8/../../../../x86_64-redhat-linux/include" "-internal-externc-isystem" "/include" "-interna/sw/SOURCES/O2/10812-slc8_x86-64/0/GPU/GPUTracking/display/GPUDisplay.cxx:36:10: error: 'omp.h' file not found [clang-diagnostic-error]
/sw/SOURCES/O2/10812-slc8_x86-64/0/GPU/GPUTracking/Base/cuda/GPUReconstructionCUDAGenRTC.cu:16:10: error: 'omp.h' file not found [clang-diagnostic-error]
/sw/SOURCES/O2/10812-slc8_x86-64/0/GPU/GPUbenchmark/cuda/../Shared/Utils.h:26:10: error: 'boost/program_options.hpp' file not found [clang-diagnostic-error]
/sw/SOURCES/O2/10812-slc8_x86-64/0/GPU/GPUbenchmark/cuda/../Shared/Utils.h:26:10: error: 'boost/program_options.hpp' file not found [clang-diagnostic-error]
/sw/SOURCES/O2/10812-slc8_x86-64/0/Framework/Logger/include/Framework/Logger.h:14:10: error: 'fairlogger/Logger.h' file not found [clang-diagnostic-error]
/usr/local/cuda-12.0/include/cuda/std/detail/libcxx/include/type_traits:520:12: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-12.0/include/cuda/std/detail/libcxx/include/type_traits:1059:69: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-12.0/include/cuda/std/detail/libcxx/include/type_traits:1175:5: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-12.0/include/cuda/std/detail/libcxx/include/type_traits:1852:16: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-12.0/include/cuda/std/detail/libcxx/include/type_traits:2276:18: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-12.0/include/cuda/std/detail/libcxx/include/type_traits:4336:16: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-12.0/include/sm_20_atomic_functions.h:103:39: error: redefinition of 'atomicAdd' [clang-diagnostic-error]
/sw/SOURCES/O2/10812-slc8_x86-64/0/Framework/Logger/include/Framework/Logger.h:14:10: error: 'fairlogger/Logger.h' file not found [clang-diagnostic-error]
/usr/local/cuda-12.0/include/cuda/std/detail/libcxx/include/type_traits:520:12: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-12.0/include/cuda/std/detail/libcxx/include/type_traits:1059:69: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-12.0/include/cuda/std/detail/libcxx/include/type_traits:1175:5: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-12.0/include/cuda/std/detail/libcxx/include/type_traits:1852:16: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-12.0/include/cuda/std/detail/libcxx/include/type_traits:2276:18: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-12.0/include/cuda/std/detail/libcxx/include/type_traits:4336:16: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-12.0/include/sm_20_atomic_functions.h:103:39: error: redefinition of 'atomicAdd' [clang-diagnostic-error]
/sw/SOURCES/O2/10812-slc8_x86-64/0/GPU/GPUbenchmark/hip/../Shared/Utils.h:146:34: error: use of undeclared identifier 'int4'; did you mean 'int'? [clang-diagnostic-error]
/sw/SOURCES/O2/10812-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:64:8: error: unknown type name '__host__' [clang-diagnostic-error]
/sw/SOURCES/O2/10812-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:64:27: error: expected ';' after top level declarator [clang-diagnostic-error]
/sw/SOURCES/O2/10812-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:64:33: error: overloaded 'operator+=' must have at least one parameter of class or enumeration type [clang-diagnostic-error]
/sw/SOURCES/O2/10812-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:64:44: error: unknown type name 'int4'; did you mean 'int'? [clang-diagnostic-error]
/sw/SOURCES/O2/10812-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:64:53: error: unknown type name 'int4'; did you mean 'int'? [clang-diagnostic-error]
/sw/SOURCES/O2/10812-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:66:4: error: member reference base type 'int' is not a structure or union [clang-diagnostic-error]
/sw/SOURCES/O2/10812-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:66:11: error: member reference base type 'int' is not a structure or union [clang-diagnostic-error]
/sw/SOURCES/O2/10812-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:67:4: error: member reference base type 'int' is not a structure or union [clang-diagnostic-error]
/sw/SOURCES/O2/10812-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:67:11: error: member reference base type 'int' is not a structure or union [clang-diagnostic-error]
/sw/SOURCES/O2/10812-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:68:4: error: member reference base type 'int' is not a structure or union [clang-diagnostic-error]
/sw/SOURCES/O2/10812-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:68:11: error: member reference base type 'int' is not a structure or union [clang-diagnostic-error]
/sw/SOURCES/O2/10812-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:69:4: error: member reference base type 'int' is not a structure or union [clang-diagnostic-error]
/sw/SOURCES/O2/10812-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:69:11: error: member reference base type 'int' is not a structure or union [clang-diagnostic-error]
/sw/SOURCES/O2/10812-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:85:1: error: unknown type name '__global__' [clang-diagnostic-error]
/sw/SOURCES/O2/10812-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:90:19: error: use of undeclared identifier 'blockIdx' [clang-diagnostic-error]
/sw/SOURCES/O2/10812-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:90:32: error: use of undeclared identifier 'blockDim' [clang-diagnostic-error]
/sw/SOURCES/O2/10812-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:90:45: error: use of undeclared identifier 'threadIdx' [clang-diagnostic-error]
/sw/SOURCES/O2/10812-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:90:78: error: use of undeclared identifier 'blockDim' [clang-diagnostic-error]
/sw/SOURCES/O2/10812-slc8_x86-64/0/GPU/GPUbenchmark/hip/../Shared/Utils.h:146:34: error: use of undeclared identifier 'int4'; did you mean 'int'? [clang-diagnostic-error]
/sw/SOURCES/O2/10812-slc8_x86-64/0/GPU/GPUbenchmark/hip/benchmark.hip.cxx:199:35: error: use of undeclared identifier 'int4'; did you mean 'int'? [clang-diagnostic-error]
/sw/slc8_x86-64/GCC-Toolchain/v12.2.0-alice1-2/lib/gcc/x86_64-unknown-linux-gnu/12.2.0/../../../../include/c++/12.2.0/bits/shared_ptr_base.h:196:22: error: expected expression [clang-diagnostic-error]
/sw/slc8_x86-64/GCC-Toolchain/v12.2.0-alice1-2/lib/gcc/x86_64-unknown-linux-gnu/12.2.0/../../../../include/c++/12.2.0/bits/shared_ptr_base.h:196:22: error: type name does not allow function specifier to be specified [clang-diagnostic-error]
/sw/slc8_x86-64/GCC-Toolchain/v12.2.0-alice1-2/lib/gcc/x86_64-unknown-linux-gnu/12.2.0/../../../../include/c++/12.2.0/bits/shared_ptr_base.h:196:22: error: use of undeclared identifier 'noinline'; did you mean 'inline'? [clang-diagnostic-error]
/sw/SOURCES/O2/10812-slc8_x86-64/0/Detectors/EMCAL/calibration/include/EMCALCalibration/EMCALCalibExtractor.h:36:10: error: 'omp.h' file not found [clang-diagnostic-error]
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/10812/da37872a1d11721fadbab47ae60831b4abf8e692/build_O2_fullCI/pretty.html).

--- a/Detectors/TPC/workflow/src/TPCIntegrateClusterReaderSpec.cxx
+++ b/Detectors/TPC/workflow/src/TPCIntegrateClusterReaderSpec.cxx
@@ -44,11 +44,12 @@ class IntegratedClusterReader : public Task
 
  private:
   void connectTrees();
-  bool checkEOS(ProcessingContext& pc);
 
   int mLane = 0;
   int mNLanes = 1;
   int mChainEntry = 0;                                       ///< processed entries in the chain
+  int mFirstTF{0};                                           ///< first TF to process
+  int mLastTF{-1};                                           ///< last TF to process
   std::unique_ptr<TChain> mChain;                            ///< input TChain
   std::vector<std::string> mFileNames;                       ///< input files
   ITPCC mTPCC, *mTPCCPtr = &mTPCC;                           ///< branch integrated number of cluster TPC currents
@@ -58,6 +59,8 @@ class IntegratedClusterReader : public Task
 
 void IntegratedClusterReader::init(InitContext& ic)
 {
+  mFirstTF = ic.options().get<int>("firstTF");
+  mLastTF = ic.options().get<int>("lastTF");
   mLane = ic.services().get<const o2::framework::DeviceSpec>().inputTimesliceId;
   mChainEntry = mLane;
   mNLanes = ic.services().get<const o2::framework::DeviceSpec>().maxInputTimeslices;
@@ -108,15 +111,36 @@ void IntegratedClusterReader::run(ProcessingContext& pc)
     // disable all branches except the firstTForbit branch to significantly speed up the loop over the TTree
     mChain->SetBranchStatus("*", 0);
     mChain->SetBranchStatus("firstTForbit", 1);
+    uint32_t countTFs = mChain->GetEntries();
+    if (mLastTF != -1) {
+      countTFs = 0;
+      mChain->SetBranchStatus("tfCounter", 1);
+    }
     for (unsigned long i = 0; i < mChain->GetEntries(); i++) {
       mChain->GetEntry(i);
       mIndices.emplace_back(std::make_pair(mTFinfo.firstTForbit, i));
+
+      if ((mLastTF != -1) && (mTFinfo.tfCounter <= mLastTF)) {
+        if ((mTFinfo.tfCounter >= mFirstTF)) {
+          // count number of TFs to process
+          ++countTFs;
+        } else {
+          // keep track of first entry in the chain
+          ++mChainEntry;
+        }
+      }
     }
     mChain->SetBranchStatus("*", 1);
     std::sort(mIndices.begin(), mIndices.end());
-    if (checkEOS(pc)) {
-      return;
-    }
+    mLastTF = countTFs;
+    LOGP(info, "Processing {} TFs out ouf {} TFs with first index in chain {}", mLastTF, mChain->GetEntries(), mChainEntry);
+  }
+
+  if (mChainEntry >= mChain->GetEntries() || (mLastTF != -1 && (pc.services().get<o2::framework::TimingInfo>().tfCounter >= mLastTF))) {
+    LOGP(info, "Quit");
+    pc.services().get<ControlService>().endOfStream();
+    pc.services().get<ControlService>().readyToQuit(QuitRequest::Me);
+    return;
   }
 
   LOGP(debug, "Processing entry {}", mIndices[mChainEntry].second);
@@ -132,8 +156,6 @@ void IntegratedClusterReader::run(ProcessingContext& pc)
 
   pc.outputs().snapshot(Output{header::gDataOriginTPC, getDataDescriptionTPCC()}, mTPCC);
   usleep(100);
-
-  checkEOS(pc);
 }
 
 void IntegratedClusterReader::connectTrees()
@@ -146,16 +168,6 @@ void IntegratedClusterReader::connectTrees()
   assert(mChain->GetEntries());
   mChain->SetBranchAddress("ITPCC", &mTPCCPtr);
   mChain->SetBranchAddress("tfID", &mTFinfoPtr);
-}
-
-bool IntegratedClusterReader::checkEOS(ProcessingContext& pc)
-{
-  if (mChainEntry >= mChain->GetEntries()) {
-    pc.services().get<ControlService>().endOfStream();
-    pc.services().get<ControlService>().readyToQuit(QuitRequest::Me);
-    return true;
-  }
-  return false;
 }
 
 DataProcessorSpec getTPCIntegrateClusterReaderSpec()
@@ -172,6 +184,8 @@ DataProcessorSpec getTPCIntegrateClusterReaderSpec()
       {"tpc-currents-infiles", VariantType::String, "o2currents_tpc.root", {"comma-separated list of input files or .txt file containing list of input files"}},
       {"input-dir", VariantType::String, "none", {"Input directory"}},
       {"dont-check-file-access", VariantType::Bool, false, {"Deactivate check if all files are accessible before adding them to the list of files"}},
+      {"firstTF", VariantType::Int, 0, {"First TF to process"}},
+      {"lastTF", VariantType::Int, -1, {"Last TF to process"}},
     }};
 }
 

--- a/Detectors/TPC/workflow/src/cluster-integrator-reader.cxx
+++ b/Detectors/TPC/workflow/src/cluster-integrator-reader.cxx
@@ -19,7 +19,7 @@ void customize(std::vector<o2::framework::ConfigParamSpec>& workflowOptions)
 {
   std::vector<ConfigParamSpec> options{
     {"time-lanes", VariantType::Int, 1, {"Number of parallel processing lanes."}},
-  };
+    ConfigParamSpec{"configKeyValues", VariantType::String, "", {"Semicolon separated key=value strings"}}};
   std::swap(workflowOptions, options);
 }
 
@@ -27,6 +27,7 @@ void customize(std::vector<o2::framework::ConfigParamSpec>& workflowOptions)
 
 WorkflowSpec defineDataProcessing(ConfigContext const& cfgc)
 {
+  o2::conf::ConfigurableParam::updateFromString(cfgc.options().get<std::string>("configKeyValues"));
   const auto time_lanes = cfgc.options().get<int>("time-lanes");
   WorkflowSpec wf;
   wf.emplace_back(timePipeline(o2::tpc::getTPCIntegrateClusterReaderSpec(), time_lanes));

